test/integration/base_integration_test.h

```diff
@@ -189,12 +189,17 @@ class BaseIntegrationTest : protected Logger::Loggable<Logger::Id::testing> {
   template <class T>
   void sendDiscoveryResponse(const std::string& type_url, const std::vector<T>& state_of_the_world,
                              const std::vector<T>& added_or_updated,
-                             const std::vector<std::string>& removed, const std::string& version) {
+                             const std::vector<std::string>& removed, const std::string& version,
+                             bool use_wrapper = false,
```

Comment:
What's the added benefit of `use_wrapper` being set to `true` without the following <key,value>?
If there isn't a need, I suggest removing the parameter, and just using the passed <key,value> as an indication to whether it needs to be wrapped or not.

Reply:
I don't think this adds any value for integration tests. I suggest removing it to reduce confusion and improve usability.

Reply:
Removed but it may be valuable to only test the decoding of the wrapper.